src/v/rpc/connection_cache.h

```diff
@@ -102,11 +102,22 @@ inline ss::shard_id connection_cache::shard_for(
   ss::shard_id src_shard,
   model::node_id n,
   ss::shard_id total_shards) {
-    static const constexpr size_t vnodes = 3;
+    if (ss::smp::count <= 8) {
+        return src_shard;
+    }
+    static const constexpr size_t vnodes = 8;
     /// make deterministic - choose 1 prime to mix node_id with
     /// https://planetmath.org/goodhashtableprimes
     static const constexpr std::array<size_t, vnodes> universe{
-      {402653189, 805306457, 1610612741}};
+      {25165843,
```

Comment:
This number is duplicated; is that intentional?

Reply:
no that's a bug.